integration/simulation/transaction_injector.go

```diff
@@ -150,6 +155,55 @@ func (ti *TransactionInjector) Stop() {
 	}
 }
 
+// issueRandomValueTransfers creates and issues a number of L2 value transfer transactions proportional to the simulation time, such that they can be processed
+func (ti *TransactionInjector) issueRandomValueTransfers() {
+	for txCounter := 0; ti.shouldKeepIssuing(txCounter); txCounter++ {
+		fromWallet := ti.rndObsWallet()
+		toWallet := ti.rndObsWallet()
+		obscuroClient := ti.rpcHandles.ObscuroWalletRndClient(fromWallet)
+		// We avoid transfers to self, unless there is only a single L2 wallet.
+		for len(ti.wallets.SimObsWallets) > 1 && fromWallet.Address().Hex() == toWallet.Address().Hex() {
```

Comment:
Should we really avoid this? I'm not sure how ethereum should handle those transactions, but either way I feel like if this must fail, it must be because of obscuro reverting the transaction rather than the test driver.

Reply:
Nah.. we shouldn't avoid this. It's just repeating the tx.
I think it might screw up the validation if we don't though..